layered-architectures-module-2/Catalog.WebUI/Controllers/ProductsController.cs

```diff
@@ -14,6 +15,13 @@ public async Task<ActionResult<ProductsVm>> GetProducts(int? id = null)
         var query = new GetProductsQuery() { Id = id };
         return await Mediator.Send(query);
     }
+    
+    [HttpGet("paginated")]
+    public async Task<ActionResult<PaginatedList<ProductDto>>> GetProductsByCategoryIdPaginated([FromQuery]int categoryId, [FromQuery]int pageNumber, [FromQuery]int pageSize)
```

Comment:
**Critical for prod issue:** there is no param validations. This may cause a DDOS

Reply:
Fixed, thanks